applications/nrf_desktop/doc/config_channel.rst

```diff
@@ -92,184 +92,233 @@ The data format is described by the following table.
 
-1. Use ``CONFIG_STATUS_INDEX_PEERS`` request to assign recipient IDs to the connected peers and make them available for configuration.
-2. Use ``CONFIG_STATUS_GET_PEER`` request to iteravitely get information about peripherals connected to the dongle over Bluetooth LE.
-3. Follow the same procedure as described in :ref:`directly connected device discovery section <nrf_desktop_config_channel_direct_device_discovery>` to discover the peripheral performing HID feature report set and get operations on the dongle, but using recipient ID that was assigned to a given peripheral when the dongle was discovered.
-   The dongle would act as a bridge between the host and the configured peripheral.
 
-If a device connected via dongle disconnects, you must discover devices connected via the dongle again to regain ability to configure it.
-The recipient ID that used to be assigned to the device before it disconnected is no longer valid.
-Discovering devices connected via the dongle is also needed to access devices that connected to the dongle after peers were indexed.
+The assigned Recipient ID is valid only during connection time between the remote device and the dongle.
+If a remote device disconnects from the dongle, the recipient discovery must be repeated when device reconnects.
 
 .. tip::
-   See :ref:`nrf_desktop_config_channel_script` for implementation example of host tools.
+   The recipient discovery is not needed if host does not interact with remote devices connected through dongle.
+
+.. _nrf_desktop_config_channel_device_identification:
+
+Device identification
+---------------------
+
+The following requests provide the information about the device hardware.
+Use the Recipient ID linked with the device that is being discovered.
+
+1. Read board name using ``CONFIG_STATUS_GET_BOARD_NAME`` request.
+#. Read hardware ID using ``CONFIG_STATUS_GET_HWID`` request.
+
+The Hardware ID allows to differentiate devices of the same type (having the same board name).
+The tool should note which Recipient ID is associated with the board name and Hardware ID.
+
+.. _nrf_desktop_config_channel_module_discovery:
+
+Module discovery
+----------------
+
+The host can access options provided by the application modules.
+Before the options are accessed the host must identified which modules are available and what options are provided by them.
+Both modules and their options are identified by name, sent as a string during discovery procedure.
+The module of given name is linked with specific value of Module ID.
+The option of given name is linked with specific value of Option ID.
+The Module ID and Option ID associated with module and option name may vary between devices, but do not change between connections to the same device and firmware version.
+
+The module discovery procedure consists of a following steps:
+
+1. Obtain the number of configurable application modules using ``CONFIG_STATUS_GET_MAX_MOD_ID`` request.
```

Comment:
That is max module ID, so number of configurable modules is greater by one than that